pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionConfigUtils.java

```diff
@@ -142,6 +142,8 @@ public static FunctionDetails convert(FunctionConfig functionConfig, ClassLoader
         Function.SubscriptionType subType = ((functionConfig.getRetainOrdering() != null && functionConfig.getRetainOrdering())
                 || FunctionConfig.ProcessingGuarantees.EFFECTIVELY_ONCE.equals(functionConfig.getProcessingGuarantees()))
                 ? Function.SubscriptionType.FAILOVER
+                : (functionConfig.getRetainKeyOrdering() != null && functionConfig.getRetainKeyOrdering())
```

Comment:
Can we re-organize this giant conditional statement to be more readable?

if (effectively-once) {
   subType = Failover
} else if (retain order) {
   subType = failover
} else if (retain key order) {
 subType = shared_key
} else {
  subType = shared
}